pydicom/pixel_data_handlers/gdcm_handler.py

```diff
@@ -30,7 +61,16 @@ def get_pixeldata(dicom_dataset):
         raise ImportError(msg)
 
     gdcm_image_reader = gdcm.ImageReader()
-    gdcm_image_reader.SetFileName(dicom_dataset.filename)
+    try:
+        if isinstance(dicom_dataset.filename, unicode):
+            gdcm_image_reader.SetFileName(
```

Comment:
Also (as @darcymason reminded me in another review), it will be easier that way to find all Python2 specific code (though I personally doubt that dropping Python 2 will be on the agenda in the next 5 years).